XenAdmin/Wizards/NewSRWizard_Pages/Frontends/LVMoISCSI.cs

```diff
@@ -473,12 +473,12 @@ private void scanTargetHostButton_Click(object sender, EventArgs e)
             if (IscsiUseChapCheckBox.Checked)
             {
                 IscsiPopulateIqnsAction = new ISCSIPopulateIQNsAction(Connection,
-                    getIscsiHost(), getIscsiPort(), IScsiChapUserTextBox.Text, IScsiChapSecretTextBox.Text);
+                    getIscsiHost(), getIscsiPort(), IScsiChapUserTextBox.Text, IScsiChapSecretTextBox.Text, SrType);

```

Comment:
Given the `Run()` methods for `lvmoiscsi` and `gfs2` don't have much in common, you could alternatively derive a new class `Gfs2PopulateIQNsAction` from `ISCSIPopulateIQNsAction` in the same file, and put the content of DoProbeExt in Run, in which case you call them from here as follows:
```
var chapUser = IscsiUseChapCheckBox.Checked ? IScsiChapUserTextBox.Text : null;
var chapPwd = IscsiUseChapCheckBox.Checked ? IScsiChapSecretTextBox.Text : null;
IscsiPopulateIqnsAction = SrType == SR.SRTypes.gfs2
    ? new Gfs2PopulateIQNsAction(Connection, getIscsiHost(), getIscsiPort(), chapUser, chapPwd)
    : new ISCSIPopulateIQNsAction(Connection, getIscsiHost(), getIscsiPort(), chapUser, chapPwd);
```
and you don''t need to change the  `ISCSIPopulateIQNsAction` constructor. Same for the Lun actions.